pkg/operator/quorumguardcontroller/quorumguardcontroller.go

```diff
@@ -1,298 +0,0 @@
-	}
-
-	var err error
-	c.clusterTopology, err = ceohelpers.GetControlPlaneTopology(c.infrastructureLister)
-	if err != nil {
-		klog.Errorf("Failed to get topology mode %w ", err)
-		return "", err
-	}
-
-	klog.Infof("HA mode is: %s", c.clusterTopology)
-
-	return c.clusterTopology, nil
-}
-
-func (c *QuorumGuardController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	err := c.ensureEtcdGuard(ctx, syncCtx.Recorder())
-	if err != nil {
-		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
-			Type:    "QuorumGuardControllerDegraded",
-			Status:  operatorv1.ConditionTrue,
-			Reason:  "Error",
-			Message: err.Error(),
-		}))
-		if updateErr != nil {
-			syncCtx.Recorder().Warning("QuorumGuardControllerUpdatingStatus", updateErr.Error())
-		}
-		return err
-	}
-
-	_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient,
-		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
-			Type:   "QuorumGuardControllerDegraded",
-			Status: operatorv1.ConditionFalse,
-			Reason: "AsExpected",
-		}))
-	return updateErr
-}
-
-func (c *QuorumGuardController) ensureEtcdGuard(ctx context.Context, recorder events.Recorder) error {
-	haTopologyMode, err := c.getTopologyMode()
-	if err != nil {
-		return err
-	}
-
-	if haTopologyMode != configv1.HighlyAvailableTopologyMode {
-		return nil
-	}
-
-	replicaCount, err := c.getMastersReplicaCount(ctx)
-	if err != nil {
-		return err
-	}
-
-	if err := c.ensureEtcdGuardDeployment(ctx, replicaCount, recorder); err != nil {
-		return err
-	}
-
-	if err := c.ensureEtcdGuardPDB(ctx, recorder); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// ensureEtcdGuardDeployment if etcd quorum guard deployment doesn't exist or was changed - apply it
-func (c *QuorumGuardController) ensureEtcdGuardDeployment(ctx context.Context, replicaCount int32, recorder events.Recorder) error {
-
-	if c.etcdQuorumGuard == nil {
-		c.etcdQuorumGuard = resourceread.ReadDeploymentV1OrDie(etcd_assets.MustAsset("etcd/quorumguard-deployment.yaml"))
-	}
-	c.etcdQuorumGuard.Spec.Replicas = &replicaCount
-	// use image from release payload
-	c.etcdQuorumGuard.Spec.Template.Spec.Containers[0].Image = c.cliImagePullSpec
-
-	// if restart occurred, we will apply etcd guard deployment but if it is the same, nothing will happened
-	actual, modified, err := resourceapply.ApplyDeploymentv1(ctx, c.kubeClient.AppsV1(), c.etcdQuorumGuard)
-	if err != nil {
-		klog.Errorf("failed to verify/apply %s, error %w", EtcdGuardDeploymentName, err)
-		return err
-	}
-
-	// if deployment was modified and is not managed by CVO need to save spec from modified one
-	// cause there are some fields added on creation or apply, for example permissions on volumes
-	// and we want to save them to be able to verify that deployment was not changed
-	// if managed by CVO, delete it
-	if !c.findAndDeleteCVOManagedQuorumGuardDeployment(ctx, actual, recorder) && modified {
```

Comment:
we have CVO resource that needs deleted are we to do this manually now?

Reply:
I thought, we don't need it from 4.8 as per the comments.